app/controllers/api/v0/events_controller.rb

```diff
@@ -6,17 +6,20 @@ def create
     render(json: error, status: error.status_code) and return if error
 
     inbound_binding.events.each do |event|
-      event.data[:user_uuid] = CompactUuid.pack(current_user_uuid) if current_user_uuid
+      event.data.user_uuid = CompactUuid.pack(current_user_uuid) if current_user_uuid
 
-      if event.schema_version > 0
-        avro_encoded_data = KafkaAvroTurf.instance.encode(event.data, schema_name: event.schema_type, version: event.schema_version)
-      else
-        avro_encoded_data = KafkaAvroTurf.instance.encode(event.data, schema_name: event.schema_type)
-      end
+      avro_schema_type = avroturf_path(data: event.data)
+      avro_encoded_data = KafkaAvroTurf.instance.encode(event.data.to_hash, schema_name: avro_schema_type, version: event.data.version)
 
       KafkaClient.async_produce(data: avro_encoded_data, topic: event.topic)
     end
 
     render nothing: true, status: 201
   end
+
+  private
+  def avroturf_path(data:)
+    split = data.type.split('.')
+    "#{data.type}.v#{data.version}.#{split.last}"
```

Comment:
If `data.type` is `'org.openstax.ec.nudged'`, wouldn't this return `'org.openstax.ec.nudged.v2.nudged'`?

Oh that's because the avsc file is named `nudged` and `nudged` is in the path.  See other comment, is that repetitive?

Also, this name goes to the [schema registry, right](https://github.com/dasch/avro_turf/blob/master/lib/avro_turf/messaging.rb#L104)?  If so, does inserting the version number mess with the registry's ability to help us with backwards and forwards compatibility management of schemas? (i.e. with the version number in the schema name won't they look like independent schemas?)

There is a [version number argument](https://github.com/dasch/avro_turf/blob/master/lib/avro_turf/messaging.rb#L98) to encode - should we be using that?

Also instead of this private method here, should we have a PORO like `AvroHash` that has a `#schema_name` method that can be unit tested?